lte/gateway/python/integ_tests/common/subscriber_db_client.py

```diff
@@ -130,7 +134,61 @@ def _get_subscriberdb_data(sid):
         lte.auth_key = bytes.fromhex(KEY)
         state = SubscriberState()
         state.lte_auth_next_seq = 1
-        return SubscriberData(sid=sub_db_sid, lte=lte, state=state)
+        if apn_list:
+            usr_prof = Non3GPPUserProfile()
+            for apn in apn_list:
+                apn_config = usr_prof.apn_config.add()
+                apn_config.service_selection = apn
+            return SubscriberData(
+                sid=sub_db_sid, lte=lte, state=state, non_3gpp=usr_prof
+            )
+        else:
+            return SubscriberData(sid=sub_db_sid, lte=lte, state=state)
+
+    @staticmethod
+    def _get_apn_data(apn):
+        """
+        Get APN data in protobuf format.
+
+        Args:
+            apn : APN/s names
+        Returns:
+            subscriber_data (protos.subscriberdb_pb2.SubscriberData)
+        """
+        # APN
+        apn_config = APNConfiguration()
+        apn_idx = 0
+        apn_config.service_selection = apn[apn_idx]
+        apn_idx += 1
+        apn_config.qos_profile.class_id = apn[apn_idx]
+        apn_idx += 1
+        apn_config.qos_profile.priority_level = apn[apn_idx]
+        apn_idx += 1
+        apn_config.qos_profile.preemption_capability = apn[apn_idx]
+        apn_idx += 1
+        apn_config.qos_profile.preemption_vulnerability = apn[apn_idx]
+        apn_idx += 1
+        apn_config.ambr.max_bandwidth_ul = apn[apn_idx]
+        apn_idx += 1
+        apn_config.ambr.max_bandwidth_dl = apn[apn_idx]
```

Comment:
This looks quite awkward. As these are positional arguments, apn_idx += 1 hides the position indices. It is better to pass apn as a dictionary for more readability.